docs/setup/upgrade/upgrade-migrations.asciidoc

```diff
@@ -55,22 +55,53 @@ This section highlights common causes of {kib} upgrade failures and how to preve
+
+Error: Unable to complete saved object migrations for the [.kibana] index. Please check the health of your Elasticsearch cluster and try again. Error: [timeout_exception]: Timed out waiting for completion of [org.elasticsearch.index.reindex.BulkByScrollTask@6a74c54]
+--------------------------------------------
 
 See https://github.com/elastic/kibana/issues/95321 for instructions to work around this issue.
  
 [float]
 ===== Corrupt saved objects
-We highly recommend testing your {kib} upgrade in a development cluster to discover and remedy problems caused by corrupt documents, especially when there are custom integrations creating saved objects in your environment. Saved objects that were corrupted through manual editing or integrations will cause migration failures with a log message like `Failed to transform document. Transform: index-pattern:7.0.0\n Doc: {...}` or `Unable to migrate the corrupt Saved Object document ...`. Corrupt documents will have to be fixed or deleted before an upgrade migration can succeed.
+We highly recommend testing your {kib} upgrade in a development cluster to discover and remedy problems caused by corrupt documents, especially when there are custom integrations creating saved objects in your environment.
+
+Saved objects that were corrupted through manual editing or integrations will cause migration failures with a log message like `Failed to transform document. Transform: index-pattern:7.0.0\n Doc: {...}` or `Unable to migrate the corrupt Saved Object document ...`. Corrupt documents will have to be fixed or deleted before an upgrade migration can succeed.
 
 For example, given the following error message:
-> Unable to migrate the corrupt saved object document with _id: 'marketing_space:dashboard:e3c5fc71-ac71-4805-bcab-2bcc9cc93275'. To allow migrations to proceed, please delete this document from the [.kibana_7.12.0_001] index.
 
-The following steps must be followed to allow the upgrade migration to succeed.
-Please be aware the Dashboard having ID `e3c5fc71-ac71-4805-bcab-2bcc9cc93275` belonging to the space `marketing_space` will no more be available:
-1. Delete the corrupt document with `DELETE .kibana_7.12.0_001/_doc/marketing_space:dashboard:e3c5fc71-ac71-4805-bcab-2bcc9cc93275`
-2. Restart {kib}
+[source,sh]
+--------------------------------------------
+Unable to migrate the corrupt saved object document with _id: 'marketing_space:dashboard:e3c5fc71-ac71-4805-bcab-2bcc9cc93275'. To allow migrations to proceed, please delete this document from the [.kibana_7.12.0_001] index.
+--------------------------------------------
+
+The following steps must be followed to allow the upgrade migration to succeed:
+
+. Remove the write block which the migration system has placed on the index:
```

Comment:
Any other context we feel is important to include here? I was deliberately keeping things as simple as possible since users probably don't/shouldn't care about how the algorithm itself works.

Reply:
We should include some statement saying that the steps we're giving here to delete the document with a corrupt id.

We're not giving an example of how to _fix_ the id and possibly retain the doc.

I'm worried that folks might already be so frustrated that they blindly follow the steps, end up with a "missing" doc and then get even more frustrated because we didn't warn them up front.

Reply:
Folks will need to be able to change `_settings` on an index to remove the write block but I'm not sure if we need to explicitly call that out.

Reply:
> Remove the write block which the migration system has placed on the index

Given that the question is often asked in SDHs, I would explicitly state `on the previous index` or `on the source index`.